csharp/extractor/Semmle.Extraction.CSharp.Standalone/AssemblyCache.cs

```diff
@@ -94,7 +94,7 @@ public AssemblyInfo ResolveReference(string id)
         {
             // Fast path if we've already seen this before.
             if (failedAssemblyInfoIds.Contains(id))
-                return AssemblyInfo.Invalid;
+                throw new AssemblyLoadException();
```

Comment:
It feels like we should have more information in this exception, for example pass `id` to the constructor of `AssemblyLoadException`.

Reply:
I can add it, but it would not be used at the moment. Previously, we had the static `AssemblyInfo.Invalid`, which also didn't have any more information. The reason why I changed it to an exception is that `AssemblyInfo.Invalid` was basically a partially constructed `AssemblyInfo`, which had a lot of uninitialized properties.